### PR TITLE
conda-store-ui 2024.10.1 + memory router

### PR DESCRIPTION
**Pull request by gabalafou**
Companion to https://github.com/conda-incubator/conda-store-ui/pull/429.

## Pull request checklist

- [x] Did you test this change locally?
- n/a Did you update the documentation (if required)?
- n/a Did you add/update relevant tests for this change (if required)?

## How to test

Load the extension locally and navigate. Make sure you can open an environment, click the edit button, click the cancel button, go to another environment, etc.

### CODE REVIEW

**Comment by gabalafou** [comment]
This a draft PR because it depends on https://github.com/conda-incubator/conda-store-ui/pull/429 getting merged and a new release of conda-store-ui getting cut. 

This PR will have to eventually update package.json with the new, fixed version of conda-store-ui.

**Comment by gabalafou** [comment]
Now that https://github.com/conda-incubator/conda-store-ui/pull/429 has been merged, we can go ahead with reviewing and merging this PR, so I'm taking it out of draft mode.

**Comment by peytondmurray** [comment]
Just tested this out locally. To be clear about installation, I'm running `docker compose up --build` inside the latest commit of `conda-store-ui` (dc1a1988b5e6212c16439ef38cc0c9be92dc8740, which includes your routing fix) to bring up conda-store at `localhost:8080/conda-store`, which I see is the default for `jupyterlab-conda-store`. Is there anything else I need to do to authenticate or anything like that?

I'm getting 404 when I try to use this:

image

File: src/widget/condaStoreWidget.tsx [hunk 1]
```diff
@@ -12,7 +12,10 @@ interface IJlabAppProps extends IAppProps {
 class JlabApp extends App<IJlabAppProps> {
   constructor({ settings }: IJlabAppProps) {
     super({
-      pref: settings.composite as IAppProps['pref'],
+      pref: {
+        ...(settings.composite as IAppProps['pref']),
+        routerType: 'memory'
```
Comment (peytondmurray) [inline src/widget/condaStoreWidget.tsx:17]: `tsc` doesn't like this: it's preventing this from building without a `// @ts-ignore`.

**Comment by trallard** [comment]
Since we have now a 2024.10.1. release for conda-store-ui can this be taken out of Draft @gabalafou ?

**Comment by gabalafou** [comment]
I updated the conda-store-ui version in this PR, so it should be possible to follow the manual testing instructions in the PR description.

I went through them myself and everything worked. 

Once @peytondmurray does the same, I think this should be good to merge.

File: src/widget/condaStoreWidget.tsx (hunk 1 above)
Comment (gabalafou) [reply src/widget/condaStoreWidget.tsx:17]: Should be good now with the update to conda-store-ui 2024.10.1, right?

**Comment by trallard** [comment]
I just tested locally, and everything worked as expected.
I only noticed that the version string went missing, but that is not a big deal.

I will leave it to @peytondmurray  to review and merge.

JupyterLab

**Review by trallard — APPROVED** [review]
LGTM so will go ahead and approve, thanks @gabalafou

**Comment by trallard** [comment]
CI failure is unrelated so will go ahead an merge (fix for CI is in #66 )

**Pull request merged**